src/strongbox/toc.clj

```diff
@@ -67,37 +69,44 @@
         contents (clojure.string/split-lines toc-contents)]
     (->> contents (filter interesting?) (map parse-comment) (reduce merge))))
 
+(defn-spec find-toc-files (s/or :ok map?, :error nil?)
+  "returns a map of {game-track filename.toc}"
+  ([addon-dir ::sp/extant-dir]
+   (find-toc-files addon-dir (fs/base-name addon-dir)))
+  ([addon-dir ::sp/extant-dir, addon-dir-name (s/nilable string?)]
+   (let [;; for cases where the .toc file doesn't match the dirname
+         addon-dir-name (or addon-dir-name ".+")
+         pattern (Pattern/compile (format "(?u)^%s(?:[\\-_](Mainline|Classic|Vanilla|TBC|BCC){1})?\\.toc$" addon-dir-name))
+         matching-toc-pattern (fn [filename]
+                                (let [toc-bname (str (fs/base-name filename))
+                                      [toc-bname-match game-track-match] (re-matches pattern toc-bname)]
+                                  (when toc-bname-match
+                                    (if-not game-track-match
+                                      [:retail toc-bname] ;; assume retail if we have a match 
```

Comment:
perhaps we should support uncertainty and go with `nil` - "we have a toc file but can't know if it's retail or something else"